File: src/csharp/Intel/Iced.UnitTests/Intel/EncoderTests/OpCodeInfoTestCases.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER && OPCODE_INFO
using System.Diagnostics;
using System.IO;
using System.Linq;

namespace Iced.UnitTests.Intel.EncoderTests {
	static class OpCodeInfoTestCases {
		public static readonly OpCodeInfoTestCase[] OpCodeInfoTests = CreateOpCodeInfos();

		static OpCodeInfoTestCase[] CreateOpCodeInfos() {
			var filename = PathUtils.GetTestTextFilename("OpCodeInfos.txt", "Encoder");
			Debug.Assert(File.Exists(filename));
			return OpCodeInfoTestCasesReader.ReadFile(filename).ToArray();
		}
	}
}
#endif

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/info/OpCodeInfoTestCases.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.info;

import com.github.icedland.iced.x86.PathUtils;

final class OpCodeInfoTestCases {
	static final OpCodeInfoTestCase[] opCodeInfoTests = createOpCodeInfos();

	private static OpCodeInfoTestCase[] createOpCodeInfos() {
		String filename = PathUtils.getTestTextFilename("Encoder", "OpCodeInfos.txt");
		return OpCodeInfoTestCasesReader.readFile(filename);
	}
}
